genesyscloud/telephony_provider_edges_trunkbasesettings/genesyscloud_telephony_providers_trunkbasesettings_proxy.go

```diff
@@ -0,0 +1,193 @@
+	}
+
+	return internalProxy
+}
+
+func (p *trunkbaseSettingProxy) GetTrunkBaseSettingById(ctx context.Context, trunkBaseSettingId string) (*platformclientv2.Trunkbase, *platformclientv2.APIResponse, error) {
+	return p.getTrunkBaseSettingByIdAttr(ctx, p, trunkBaseSettingId)
+}
+
+func (p *trunkbaseSettingProxy) GetAllTrunkBaseSetting(ctx context.Context) (*[]platformclientv2.Trunkbase, *platformclientv2.APIResponse, error) {
+	return p.getAllTrunkBaseSettingsAttr(ctx, p, "")
+}
+
+func (p *trunkbaseSettingProxy) GetAllTrunkBaseSettingWithName(ctx context.Context, name string) (*[]platformclientv2.Trunkbase, *platformclientv2.APIResponse, error) {
+	return p.getAllTrunkBaseSettingsAttr(ctx, p, name)
+}
+func (p *trunkbaseSettingProxy) CreateTrunkBaseSetting(ctx context.Context, trunkBaseSetting platformclientv2.Trunkbase) (*platformclientv2.Trunkbase, *platformclientv2.APIResponse, error) {
+	return p.createTrunkBaseSettingAttr(ctx, p, trunkBaseSetting)
+}
+
+func (p *trunkbaseSettingProxy) UpdateTrunkBaseSetting(ctx context.Context, trunkbaseSettingId string, trunkBaseSetting platformclientv2.Trunkbase) (*platformclientv2.Trunkbase, *platformclientv2.APIResponse, error) {
+	return p.updateTrunkBaseSettingAttr(ctx, p, trunkbaseSettingId, trunkBaseSetting)
+}
+
+func (p *trunkbaseSettingProxy) DeleteTrunkBaseSetting(ctx context.Context, trunkbaseSettingId string) (*platformclientv2.APIResponse, error) {
+	return p.deleteTrunkBaseSettingAttr(ctx, p, trunkbaseSettingId)
+}
+
+func getTrunkBaseSettingByIdFn(ctx context.Context, p *trunkbaseSettingProxy, trunkBaseSettingId string) (*platformclientv2.Trunkbase, *platformclientv2.APIResponse, error) {
+	tb := rc.GetCacheItem(p.trunkBaseCache, trunkBaseSettingId)
+	if tb != nil {
+		return tb, nil, nil
+	}
+
+	return p.edgesApi.GetTelephonyProvidersEdgesTrunkbasesetting(trunkBaseSettingId, true)
+}
+
+func getAllTrunkBaseSettingsFn(ctx context.Context, p *trunkbaseSettingProxy, name string) (*[]platformclientv2.Trunkbase, *platformclientv2.APIResponse, error) {
+	trunkbaseSlice := make([]platformclientv2.Trunkbase, 0, 10)
+	util.WithRetries(ctx, 5*time.Second, func() *retry.RetryError {
+		for pageNum := 1; ; pageNum++ {
+			const pageSize = 100.
+			trunkBaseSettings, resp, getErr := getTelephonyProvidersEdgesTrunkbasesettings(p, pageNum, pageSize, name)
+
+			if getErr != nil {
+				return retry.NonRetryableError(util.BuildWithRetriesApiDiagnosticError(resourceName, fmt.Sprintf("Error requesting trunk base settings %s | error: %s", name, getErr), resp))
+			}
+
+			if trunkBaseSettings.Entities == nil || len(*trunkBaseSettings.Entities) == 0 {
+				return retry.RetryableError(util.BuildWithRetriesApiDiagnosticError(resourceName, fmt.Sprintf("No trunkBaseSettings found with name %s", name), resp))
+			}
+
+			for _, trunkBaseSetting := range *trunkBaseSettings.Entities {
+				if trunkBaseSetting.State != nil && *trunkBaseSetting.State != "deleted" {
+					if name == "" {
```

Comment:
just a new key is added for the cache , can still set the cache even if there is specific name for fetching the trunkBaseSetting ?

Reply:
We have two different use cases.  Datasource allows you to fetch all the fields that match the name.  In that case we are not going to add them to the cache.  The get calls the same API put passes an empty name string.  This ,teams we should add the item by cache